Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,229 @@
+
+namespace Backend.Controllers
+{
+    [Route("v1/product")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly Storj storage;
+
+
+        public ProductController(Db context, ImageService imageService, Storj storage)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(Product newProduct)
+        {
+            newProduct.IsDraft = true;
+
+            db.Products.Add(newProduct);
+
+            var saved = await db.Save();
+
+            if (saved)
+            {
+                var createdProduct = await GetProduct(newProduct.Id);
+                if (createdProduct.Result is OkObjectResult okResult)
+                {
+                    return Ok(okResult.Value);
+                }
+            }
+
+            return Problem();
+        }
+
+    
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Product>> GetProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(product);
+        }
+
+   
+        [HttpPatch("{id}")]
+        public async Task<IActionResult> UpdateProduct([FromBody] Product patchDoc)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == patchDoc.Id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+
+            if (patchDoc.Name != null)
+            {
+                product.Name = patchDoc.Name;
+            }
+
+            if (patchDoc.Amount != null)
+            {
+                product.Amount = patchDoc.Amount;
+            }
+
+            if (patchDoc.Price != null)
+            {
+                product.Price = patchDoc.Price;
+            }
+
+            if (patchDoc.PreviewImage != null)
+            {
+                product.PreviewImage = patchDoc.PreviewImage;
+            }
+
+            if (patchDoc.VideoUrl != null)
+            {
+                product.VideoUrl = patchDoc.VideoUrl;
+            }
+
+            if (patchDoc.SeoTitle != null)
+            {
+                product.SeoTitle = patchDoc.SeoTitle;
+            }
+
+            if (patchDoc.SeoSlug != null)
+            {
+                product.SeoSlug = patchDoc.SeoSlug;
+            }
+
+            if (patchDoc.SeoDescription != null)
+            {
+                product.SeoDescription = patchDoc.SeoDescription;
+            }
+
+            var saved = await db.Save();
+
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPut("{id}/publish")]
+        public async Task<IActionResult> PublishProduct(string id)
```

Comment:
You should check for user. Add [Authorize]. Get uid by `User.FindFirst(Jwt.Uid)`. Check that shop is owned by user.